Controls/Tests/Test_Mail.c

```diff
@@ -17,16 +17,76 @@ int main(void){
 	OS_ERR err;
 
 	OSInit(&err);	// Initialize the OS
+
 	OS_CPU_SysTickInit();
 	
 	// Create any semaphores/mutexes and initialize any global variables here
-	
+
+	// Creates Semaphor
+	OSSemCreate(&MailboxFlag_Sem4, "Mailbox Flag", 0, &err);
```

Comment:
What if the value in the semaphore was initialized to 1? `OSSemCreate(&MailboxFlag_Sem4, "Mailbox Flag", 1, &err);`

Reply:
That value represents the initial value of the semaphor, the 0 indicates it is in use at the start of the program. The recipient is waiting for a signal from the semaphor at the start of the program (a value greater than 1). So if the semaphor was initialized to 1, the mailman and recipient would run at the same time at the start of the program. This could result in no message the first time the recipient checks.